src/main/java/JsoupBadLinks.java

```diff
@@ -0,0 +1,111 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import java.net.*;
+import java.io.*;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+public class JsoupBadLinks {
+    private static final int BAD_CODE = 404;
+    private static final int BAD_CODE2 = 505;
+
+
+    public int getResponseCode(String link) {
+        URL url;
+        HttpURLConnection con = null;
+        Integer responsecode = 0;
+        try {
+            url = new URL(link);
+            con = (HttpURLConnection) url.openConnection();
+            responsecode = con.getResponseCode();
+        } catch (Exception e) {
+        } finally {
+            if (null != con)
+                con.disconnect();
+        }
+        return responsecode;
+
+    }
+
+
+    public List<String> allLinks(String url) throws IOException {
+
+        Document doc = Jsoup.connect(url).get();
+        Elements links = doc.select("a[href]");
+        List<String> allLinks = new ArrayList<String>();
+        for (Element link : links) {
+            allLinks.add(link.attr("abs:href"));
+
+        }
+        allLinks.stream()
+                .distinct();
+        return allLinks;
+    }
+
+
+    public List<String> badLinks(String url) throws IOException {
+
+        List<String> allUrls = allLinks(url);
+        List<String> badUrls = new ArrayList<String>();
+        List<String> badUrls505 = new ArrayList<String>();
+        int i = 0;
+        int b = 0;
```

Comment:
@dwaynewade0033 why do you need these values? Can you use `list.size()` instead?